### PR TITLE
Add --check-dependent-api-module to demisto-sdk lint in content build

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3458)

## Description
Add --check-dependent-api-module to demisto-sdk lint in content build.

### CODE REVIEW

**Comment by DinaMeylakh** [comment]
Tests:
I changed CoreIRAPIModule and saw that lint runs on dependencies as well:
In circleci: https://app.circleci.com/pipelines/github/demisto/content/246015/workflows/e6291510-90c0-459d-ab2e-aac472b4c3f2/jobs/693436
In gitlab: https://code.pan.run/xsoar/content/-/jobs/23447398

Runs were successful.

**Review by ShahafBenYakir — APPROVED** [review]
Looks ok on my side.
@ilaner please have look as well.

File: Packs/ContentManagement/doc_files/ci-cd.yml [hunk 1]
```diff
@@ -106,7 +106,7 @@ jobs:
           cp $GITHUB_WORKSPACE/content/Tests/scripts/dev_envs/pytest/conftest.py ./dev_envs/pytest/
 
           # Run lint on all changed files
-          demisto-sdk lint -g -vvv
+          demisto-sdk lint -g -vvv --check-dependent-api-module
```
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply Packs/ContentManagement/doc_files/ci-cd.yml]: you are right its not relevant here

File: .circleci/config.yml [hunk 2]
```diff
@@ -326,7 +331,7 @@ jobs:
 
   Run Unit Testing And Lint:
     <<: *container_config
-    resource_class: medium
+    resource_class: large
```
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply .circleci/config.yml:334]: Unfortunately yes, otherwise the step runs out of memory and receives a Kill signal.

**Review by ilaner — APPROVED** [review]
Good job :)

**Pull request merged**